starlite/middleware/compression/brotli.py

```diff
@@ -0,0 +1,147 @@
+                    quality=self.quality,
+                    mode=self.mode,
+                    lgwin=self.lgwin,
+                    lgblock=self.lgblock,
+                )
+                await responder(scope, receive, send)
+                return
+            elif self.gzip_fallback and "gzip" in headers.get("Accept-Encoding", ""):
+                responder = GZipResponder(self.app, self.minimum_size)
+                await responder(scope, receive, send)
+                return
+        await self.app(scope, receive, send)
+
+
+class BrotliResponder:
+    def __init__(
+        self,
+        app: ASGIApp,
+        minimum_size: int,
+        quality: int,
+        mode: str,
+        lgwin: int,
+        lgblock: int,
+    ) -> None:
+        self.app = app
+        self.minimum_size = minimum_size
+        self.send: Send = unattached_send
+        self.initial_message: Message = {}
+        self.started = False
+        self.br_buffer = io.BytesIO()
+        self.quality = quality
+        self.mode = mode
+        self.lgwin = lgwin
+        self.lgblock = lgblock
+        self.br_file = brotli.Compressor(quality=self.quality, mode=self.mode, lgwin=self.lgwin, lgblock=self.lgblock)
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        self.send = send
+        await self.app(scope, receive, self.send_with_brotli)
+
+    async def send_with_brotli(self, message: Message) -> None:
+        """Handles and compresses the HTTP Message with brotli"""
+        message_type = message["type"]
+        if message_type == "http.response.start":
+            # Don't send the initial message until we've determined how to
+            # modify the outgoing headers correctly.
+            self.initial_message = message
+        elif message_type == "http.response.body" and not self.started:
+            self.started = True
+            body = message.get("body", b"")
+            more_body = message.get("more_body", False)
+            if len(body) < self.minimum_size and not more_body:
+                # Don't apply Brotli to small outgoing responses.
+                await self.send(self.initial_message)
+                await self.send(message)
+            elif not more_body:
+                # Standard Brotli response.
+                body = self.br_file.compress(body) + self.br_file.finish()
+                headers = MutableHeaders(raw=self.initial_message["headers"])
+                headers["Content-Encoding"] = "br"
+                headers["Content-Length"] = str(len(body))
+                headers.add_vary_header("Accept-Encoding")
+                message["body"] = body
+
+                await self.send(self.initial_message)
+                await self.send(message)
+            else:
+                # Initial body in streaming Brotli response.
+                headers = MutableHeaders(raw=self.initial_message["headers"])
+                headers["Content-Encoding"] = "br"
```

Comment:
"br" should be a constants at the top of the file, and also be used in the tests.